### PR TITLE
asset-swapper: Improve logging for 0x API consumption

### PR DESCRIPTION
**Pull request by feuGeneA**
## Description

Please review individual commit messages.

## Testing instructions

Run automated (mocha) tests and view logging output.

## Types of changes

* New feature (non-breaking change which adds functionality)

### CODE REVIEW

File: packages/asset-swapper/src/utils/quote_requestor.ts [hunk 1]
```diff
@@ -331,4 +271,71 @@ export class QuoteRequestor {
...
+                        };
+                        this._warningLogger(
+                            convertIfAxiosError(err),
+                            `Failed to get RFQ-T ${quoteType} quote from market maker endpoint ${url} for API key ${
+                                options.apiKey
+                            } for taker address ${options.takerAddress}`,
+                        );
+                        return undefined;
+                    }
+                    this._infoLogger({ rfqtMakerResponse: { ...logEntry, makerEndpoint: url, quoteType } });
```
Comment (feuGeneA) [inline packages/asset-swapper/src/utils/quote_requestor.ts:329]: This is broken and will be fixed shortly.  Note that this log emission happens after the return statement(s) above, so it has no effect!

File: packages/asset-swapper/src/utils/quote_requestor.ts [hunk 2]
```diff
@@ -331,4 +271,71 @@ export class QuoteRequestor {
...
+                        const response = await Axios.get<ResponseT>(`${url}/${quotePath}`, {
+                            headers: { '0x-api-key': options.apiKey },
+                            params: {
+                                takerAddress: options.takerAddress,
+                                ...inferQueryParams(marketOperation, makerAssetData, takerAssetData, assetFillAmount),
+                            },
+                            timeout: options.makerEndpointMaxResponseTimeMs,
+                        });
+                        logEntry = {
+                            statusCode: response.status,
```
Comment (PirosB3) [inline packages/asset-swapper/src/utils/quote_requestor.ts]: to we want to include `quoteType` here now?

File: packages/asset-swapper/src/utils/quote_requestor.ts [hunk 3]
```diff
@@ -80,13 +80,25 @@ function hasExpectedAssetData(
...
 
+function convertIfAxiosError(error: any): Error {
+    if (error.hasOwnProperty('isAxiosError') && error.isAxiosError && error.hasOwnProperty('toJSON')) {
+        return error.toJSON();
+    } else {
+        return error;
+    }
+}
+
+export type LogFunction = (obj: object, msg?: string, ...args: any[]) => void;
```
[the remark this answers is not in the document]
Comment (feuGeneA) [reply packages/asset-swapper/src/utils/quote_requestor.ts:91]: No, but I don't think that matters for our purposes here.  Any time we output a message, we're also outputting an object.  If that situation changes for some reason, we can change this at that point.

File: packages/asset-swapper/src/utils/quote_requestor.ts [hunk 4]
```diff
@@ -80,13 +80,25 @@ function hasExpectedAssetData(
     return hasExpectedMakerAssetData && hasExpectedTakerAssetData;
 }
 
+function convertIfAxiosError(error: any): Error {
```
[the remark this answers is not in the document]
Comment (feuGeneA) [reply packages/asset-swapper/src/utils/quote_requestor.ts]: You're right, thank you for noticing this.  I clarified it in the code here: https://github.com/0xProject/0x-monorepo/pull/2581/commits/dede076835fbc48dd1a5ab7f65446f2e0d105b75

File: packages/asset-swapper/src/utils/quote_requestor.ts [hunk 5]
```diff
@@ -100,52 +112,19 @@ export class QuoteRequestor {
...
+            'firm',
         );
 
-        const ordersWithStringInts = responses.map(response => response.data); // not yet BigNumber
-
         const validatedOrdersWithStringInts = ordersWithStringInts.filter(order => {
             const hasValidSchema = this._schemaValidator.isValid(order, schemas.signedOrderSchema);
             if (!hasValidSchema) {
-                this._warningLogger(`Invalid RFQ-t order received, filtering out: ${JSON.stringify(order)}`);
+                this._warningLogger(order, 'Invalid RFQ-t order received, filtering out');
```
[the remark this answers is not in the document]
Comment (feuGeneA) [reply packages/asset-swapper/src/utils/quote_requestor.ts:127]: Good question.  I checked.  Seems fine:
```
Unexpected asset data in RFQ-T order, filtering out: {"makerAddress":"0x0000000000000000000000000000000000000000","makerAssetAmount":"0","takerAssetAmount":"0","makerAssetData":"0xf47261b000000000000000000000000034d402f14d58e001d8efbe6585051bf9706aa064","takerAssetData":"0xf47261b0000000000000000000000000cdb594a32b1cc3479d8746279712c39d18a07fc0","makerFeeAssetData":"0xf47261b00000000000000000000000000000000000000000000000000000000000000000","takerFeeAssetData":"0xf47261b00000000000000000000000000000000000000000000000000000000000000000","takerAddress":"0x0000000000000000000000000000000000000000","senderAddress":"0x0000000000000000000000000000000000000000","makerFee":"0","takerFee":"0","feeRecipientAddress":"0x0000000000000000000000000000000000000000","salt":"6516259636314380459489620045617303281264402028101542376173959174402001077031","expirationTimeSeconds":"1589234447","exchangeAddress":"0x0000000000000000000000000000000000000000","chainId":1337,"signature":"0x"}
```
Here's a friendlier view of that object:
image

File: packages/asset-swapper/src/utils/quote_requestor.ts (hunk 2 above)
Comment (feuGeneA) [reply packages/asset-swapper/src/utils/quote_requestor.ts]: Yup, just added that in a commit that came in I guess right after your review.  See here: https://github.com/0xProject/0x-monorepo/pull/2581/commits/c929782e0d087f2051e673d20c672478807b8dcd#diff-951722c8cfc47f2ea3eebf81342ae48dR292

File: packages/asset-swapper/src/utils/quote_requestor.ts (hunk 5 above)
[the remark this answers is not in the document]
Comment (PirosB3) [reply packages/asset-swapper/src/utils/quote_requestor.ts:127]: it looks wonderful!

**Pull request merged**